suricata.yaml.in

```diff
@@ -668,6 +668,53 @@ af-packet:
     #use-mmap: no
     #tpacket-v3: yes
 
+# Linux high speed af-xdp capture support
+af-xdp:
+  - interface: default
+    # Number of receive threads. "auto" uses least between the number
+    # of cores and RX queues
+    #threads: auto
+    #disable-promisc: false
+    # XDP_DRV mode can be chosen when the driver supports XDP
+    # XDP_SKB mode can be chosen when the driver does not support XDP
+    # Possible values are:
+    #  - drv: enable XDP_DRV mode
```

Comment:
Is hardware mode supported by some cards ?

Reply:
According to ``if_link.h`` there is a ``XDP_FLAGS_HW_MODE`` option. However, I briefly attempted this and it appears my NIC does not support this feature (may be setup related), therefore I could not test and opted to exclude it. Also the last paragraph of the overview states there are only two modes. I didn't want to diverge too much from the official documentation for an initial release, see: https://01.org/linuxgraphics/gfx-docs/drm/networking/af_xdp.html#overview

Do you have a preference here? I could provide the option, but unsure if I could test it.

Reply:
Do you happen to know which NICs (+ I guess kernel/driver versions) support it?

Reply:
Support hardware mode? I do not, currently. I can try and find that out. I refer back to the documentation where it states there are only two modes, both of which I have implemented. Do you think we should provide the option, even though it is not suggested in the documentation?

Reply:
Was thinking that we may get someone to test the mode so that we can support it.

Reply:
Yeah I agree. I opted to exclude it for the reason that I couldn't test it. And therefore when/if it were introduced in a new PR it could be tested and documented accordingly.

Reply:
I have access to netronome card that supports XDP in hardware. I could give it a try (but need to check if AF_XDP is supported).

Reply:
So would you rather I provide the ability and leave it untested? Or should I leave it as it currently is and allow you to introduce this?

Reply:
Actually I'm not entirely sure our exact interaction with the eBPF programs, maybe @rmcconnell-r7 can clarify, do we leave whatever eBPF program is in place? Or do we just pass NULL to all that to get the default/nop/just-forward-everything program?

Reply:
If the netronome nic is the only thing that might benefit (and even that seems an open question) then I'm fine with leaving that as something that can be done later, perhaps by someone from netronome if they would care enough and see a benefit somehow.